include/pybind11/pytypes.h

```diff
@@ -1784,24 +1784,41 @@ class kwargs : public dict {
     PYBIND11_OBJECT_DEFAULT(kwargs, dict, PyDict_Check)
 };
 
-class set : public object {
+class any_set : public object {
+protected:
+    PYBIND11_OBJECT(any_set, object, PyAnySet_Check)
+
+public:
+    size_t size() const { return (size_t) PySet_Size(m_ptr); }
```

Comment:
While you're at it, could you please change this to `static_cast<size_t>`? (mainly for readability)